libs/common-services/src/main/java/uk/gov/di/ipv/core/library/helpers/EmbeddedMetricHelper.java

```diff
@@ -0,0 +1,59 @@
+package uk.gov.di.ipv.core.library.helpers;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import software.amazon.cloudwatchlogs.emf.logger.MetricsLogger;
+import software.amazon.cloudwatchlogs.emf.model.DimensionSet;
+import uk.gov.di.ipv.core.library.annotations.ExcludeFromGeneratedCoverageReport;
+
+import java.util.Map;
+
+import static uk.gov.di.ipv.core.library.helpers.LogHelper.GOVUK_SIGNIN_JOURNEY_ID_DEFAULT_VALUE;
+import static uk.gov.di.ipv.core.library.helpers.LogHelper.LogField.LOG_GOVUK_SIGNIN_JOURNEY_ID;
+
+@ExcludeFromGeneratedCoverageReport
+public class EmbeddedMetricHelper {
+    private static final MetricsLogger METRICS_LOGGER = getMetricsLogger();
+    private static final String CORE_BACK_EMBEDDED_METRICS_NAMESPACE = "CoreBackEmbeddedMetrics";
+
+    @Getter
+    @AllArgsConstructor
+    public enum Metric {
+        CRI_REDIRECT("criRedirect"),
+        CRI_RETURN("criReturn");
+
+        private final String name;
+    }
+
+    @Getter
+    @AllArgsConstructor
+    public enum Dimension {
+        CRI("cri");
+
+        private final String name;
+    }
+
+    public static void createMetrics(
+            Map<Dimension, String> dimensions,
+            Map<Metric, Double> metrics,
+            String govukSigninJourneyId) {
+        METRICS_LOGGER.setNamespace(CORE_BACK_EMBEDDED_METRICS_NAMESPACE);
+        METRICS_LOGGER.putProperty(
```

Comment:
properties don't get included in the metric, so it's safe to have values with high cardinality. They are included in the log message that's created.

Reply:
I wonder if there's any value in stealing the Log4J context directly - e.g. with something like `ThreadContext.getContext()`, so you automatically get all the other common log properties?

Might risk key clashes, and Journey ID is definitely the most useful, so this is also fine!